shell/platform/common/cpp/text_input_model.cc

```diff
@@ -150,65 +143,53 @@ bool TextInputModel::DeleteSurrounding(int offset_from_cursor, int count) {
   text_.erase(start, end - start);
 
   // Cursor moves only if deleted area is before it.
-  if (offset_from_cursor <= 0) {
-    selection_base_ = start;
-  }
-
-  // Clear selection.
-  selection_extent_ = selection_base_;
+  selection_ = TextRange(offset_from_cursor <= 0 ? start : selection_.start());
 
   return true;
 }
 
 bool TextInputModel::MoveCursorToBeginning() {
-  if (selection_base_ == 0 && selection_extent_ == 0)
+  if (selection_.start() == 0 && selection_.end() == 0)
```

Comment:
I think `collapsed && position == 0` is a clearer expression of the intent here using the new API.

Reply:
Good call. Done.